src/GenAPI/Microsoft.DotNet.GenAPI/SyntaxGeneratorExtensions.cs

```diff
@@ -43,5 +64,24 @@ public static SyntaxNode DeclarationExt(this SyntaxGenerator syntaxGenerator, IS
                 throw new ArgumentException(ex.Message, symbol.ToDisplayString());
             }
         }
+
+        private static ConstructorInitializerSyntax GenerateBaseConstructorInitializer(ImmutableArray<IMethodSymbol> baseTypeConstructors)
+        {
+            ConstructorInitializerSyntax constructorInitializer = SyntaxFactory.ConstructorInitializer(SyntaxKind.BaseConstructorInitializer);
+
+            IMethodSymbol baseConstructor = baseTypeConstructors.OrderBy(c => c.Parameters.Count()).First();
```

Comment:
The legacy GenAPI handled this case as well: https://github.com/dotnet/arcade/blob/608f3c7aed68dec1314d75601c6f143747e78cc6/src/Microsoft.Cci.Extensions/Writers/CSharp/CSDeclarationWriter.Methods.cs#L444-L445

Do we want to port that over here as well?

Reply:
Unsure but can we just mimic the existing legacy GenAPI behavior? cc @ericstj

Reply:
I'd start by saying that GenAPI doesn't need to handle a case that's impossible to represent in the product.  It also shouldn't go out of its way to support something that's extremely unlikely (eg: we Obsolete(true) an API that is in active use by some up-stack component).

Worst case, if we somehow had a case of this scenario, then I would expect that we'd need to workaround it by having an internal constructor on the base type. Manual exclusion could also allow a user to exclude the upstack constructor if they needed to redefine it to call a non-default internal constructor.  I would recommend that GenAPI omit the base call to an obsolete(true) constructor.  This is what the [old GenAPI did](https://github.com/dotnet/arcade/blob/608f3c7aed68dec1314d75601c6f143747e78cc6/src/Microsoft.Cci.Extensions/Writers/CSharp/CSDeclarationWriter.Methods.cs#L444-L445).  

TL;DR
1. Omit all calls to base obsolete(true) constructors.
2. Developers can author an internal constructor on the base type manually (and a call it, if it's not default).